interface/external/pkmn_external.go

```diff
@@ -0,0 +1,163 @@
+}
+
+func NewPokemonExternal() PokemonExternal {
+	return &pokemonExternal{}
+}
+
+func (pe *pokemonExternal) GetExternalPokemon(p []*model.Pokemon, c context.Context) ([]*model.Pokemon, error) {
+	request, err := http.Get(fmt.Sprintf("%v?limit=50", url))
+
+	if err != nil {
+		u.ErrorLogger.Println("External resource is not reachable")
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(request.Body)
+
+	if err != nil {
+		u.ErrorLogger.Println("Response body could not be read")
+		return nil, err
+	}
+
+	response := external.PokemonExternalResponse{}
+
+	if err := json.Unmarshal(body, &response); err != nil {
+		return nil, err
+	}
+
+	for i, val := range response.Results {
+		id := uint64(i + 1)
+
+		details_request, err := http.Get(fmt.Sprintf("https://pokeapi.co/api/v2/pokemon/%v/", id))
+
+		if err != nil {
+			u.ErrorLogger.Printf("External details are not reachable: %s", err)
+			return nil, err
+		}
+
+		details_body, err := ioutil.ReadAll(details_request.Body)
+
+		if err != nil {
+			u.ErrorLogger.Println("Details body could not be read")
+			return nil, err
+		}
+
+		details_response := external.PokemonExternalDetails{}
+
+		if err := json.Unmarshal(details_body, &details_response); err != nil {
+			u.ErrorLogger.Printf("Details could not be unmarshalled: %v", err)
+			return nil, err
+		}
+
+		if len(details_response.Types) < 2 {
+			pkmn := &model.Pokemon{
+				ID:          id,
+				Name:        val.Name,
+				Generation:  getGeneration(id),
+				PrimaryType: details_response.Types[0].TypeStructure.TypeName,
+				Stats: model.Stats{
+					HP:             details_response.Stats[0].BaseStat,
+					Attack:         details_response.Stats[1].BaseStat,
+					Defense:        details_response.Stats[2].BaseStat,
+					SpecialAttack:  details_response.Stats[3].BaseStat,
+					SpecialDefense: details_response.Stats[4].BaseStat,
+					Speed:          details_response.Stats[5].BaseStat,
+					BaseStatTotal:  getBST(details_response.Stats),
+				},
+			}
+			u.GeneralLogger.Println(pkmn)
+			p = append(p, pkmn)
+		} else {
+			pkmn := &model.Pokemon{
+				ID:            id,
+				Name:          val.Name,
+				Generation:    getGeneration(id),
+				PrimaryType:   details_response.Types[0].TypeStructure.TypeName,
+				SecondaryType: details_response.Types[1].TypeStructure.TypeName,
+				Stats: model.Stats{
+					HP:             details_response.Stats[0].BaseStat,
+					Attack:         details_response.Stats[1].BaseStat,
+					Defense:        details_response.Stats[2].BaseStat,
+					SpecialAttack:  details_response.Stats[3].BaseStat,
+					SpecialDefense: details_response.Stats[4].BaseStat,
+					Speed:          details_response.Stats[5].BaseStat,
+					BaseStatTotal:  getBST(details_response.Stats),
+				},
+			}
+			u.GeneralLogger.Println(pkmn)
+			p = append(p, pkmn)
+		}
+	}
+	return p, nil
+}
+
+func (pe *pokemonExternal) SaveExternalPokemon(p []*model.Pokemon) {
+	csvFile, err := os.Create("pkmn.csv")
```

Comment:
I can't remember, but didn't you already have a package that handles csv operations? Would be better to just keep the logic that communicates with an external api here, and another package that handles every operation you could do with csv's.

That way we can have things properly separated

Also don't hardcode the csv path, we want to keep it flexible enough to be able to test this with other files